src/nbl/video/utilities/CComputeBlit.cpp

```diff
@@ -0,0 +1,152 @@
+
+	auto gpuUnspecShader = device->createGPUShader(std::move(cpuShaderOverriden));
+
+	return device->createGPUSpecializedShader(gpuUnspecShader.get(), { nullptr, nullptr, "main" });
+}
+
+core::smart_refctd_ptr<video::IGPUSpecializedShader> CComputeBlit::createBlitSpecializedShader(const asset::E_FORMAT inImageFormat, const asset::E_FORMAT outImageFormat,
+	const asset::E_FORMAT outImageViewFormat, const asset::IImage::E_TYPE inImageType, const core::vectorSIMDu32& inExtent, const core::vectorSIMDu32& outExtent,
+	const asset::IBlitUtilities::E_ALPHA_SEMANTIC alphaSemantic, const uint32_t workgroupSize, const uint32_t alphaBinCount)
+{
+	char shaderStart[] = "#version 460 core\n#include \"nbl/builtin/glsl/blit/default_compute_blit.comp\"\n";
+	auto shader = core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>>>(strlen(shaderStart), shaderStart, core::adopt_memory);
+
+	auto cpuShader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(shader), asset::ICPUShader::buffer_contains_glsl, asset::IShader::ESS_COMPUTE, "CComputeBlit::createBlitSpecializedShader");
+	
+	const core::vectorSIMDu32 windowDim = getWindowDim(inExtent, outExtent);
+
+	// Fail if any dimension of the window is bigger than workgroup size, forcing us to reuse inovcation to process just a single dimension of that window
+	if ((windowDim.x > workgroupSize) || (windowDim.y > workgroupSize) || (windowDim.z > workgroupSize))
+		return nullptr;
+
+	const uint32_t windowPixelCount = windowDim.x * windowDim.y * windowDim.z;
+	// Fail if I would need to reuse invocations just to process a single window
+	if (windowPixelCount > workgroupSize)
+		return nullptr;
+
+	const uint32_t outChannelCount = asset::getFormatChannelCount(outImageFormat);
+	// It is important not to use asset::getTexelOrBlockBytesize here.
+	// Even though input pixels are stored in the shared memory we use outChannelCount here because only outChannelCount channels of
+	// the input image actually need blitting and hence only those channels are stored in shared memory.
+	const size_t windowSize = static_cast<size_t>(windowPixelCount) * outChannelCount * sizeof(float);
+	// Fail if the window cannot be preloaded into shared memory
+	// Todo(achal): Take smem size as a param of this function?
+	if (windowSize > sharedMemorySize)
+		return nullptr;
+
+	// Todo(achal): All this belongs in validation
+	{
+		const uint32_t inChannelCount = asset::getFormatChannelCount(inImageFormat);
+		assert(outChannelCount <= inChannelCount);
+
+		// inFormat should support SAMPLED_BIT format feature
+	}
+
+	const char* outImageFormatGLSLString = getGLSLFormatStringFromFormat(outImageFormat);
+	const char* glslFormatQualifier = getGLSLFormatStringFromFormat(outImageViewFormat);
+
+	char formatInclude[1024] = "";
+	if (outImageFormat != outImageViewFormat)
+		snprintf(formatInclude, sizeof(formatInclude), "#include <nbl/builtin/glsl/blit/blit/formats/%s.glsl>\n", outImageFormatGLSLString);
```

Comment:
already marked as a TODO to remove, but just reminding again

Reply:
Resolved.